lib/command.ts

```diff
@@ -638,17 +638,31 @@ export class CommandInstance {
           // any new aliases need to be placed in positionalMap, which
           // is used for validation.
           if (!positionalMap[key]) positionalMap[key] = parsed.argv[key];
+
           // Addresses: https://github.com/yargs/yargs/issues/1637
-          // If both positionals/options provided,
-          // and no default or config values were set for that key,
-          // and if at least one is an array: don't overwrite, combine.
+          // In most cases, positionals will overwrite existing values on argv.
+          // They will be combined if both positionals/options are provided, and at least one is array type
           if (
-            !this.isInConfigs(yargs, key) &&
```

Comment:
These two lines are problematic:
if a key has default values, they will be excluded. This means that option values will be excluded when used with default values. The same will happen with config values (which is only okay when `combine-arrays` is false). 

This is the bug you saw the other day